hclpack/structure.go

```diff
@@ -101,6 +101,12 @@ func (b *Body) content(schema *hcl.BodySchema, remain *Body) (*hcl.BodyContent,
 				}
 			}
 		}
+
+		// Redeclare attribute so it doesn't get overwritten in the next
+		// iteration of the for loop, causing the diagnostic NameRange to
+		// be overwritten.
+		attr := attr
```

Comment:
Good catch! Thanks @akupila.

A different way to do this which I think would be easier to read is to change the `Subject` below like this:

```go
    Subject: attr.NameRange.Ptr(),
```

That `Ptr` method returns a pointer to its receiver, but because its receiver type is not itself a pointer it creates a copy first to pass it in. That's been the most common way to get the right behavior in these cases when it's only the diagnostics that need the distinct variable.

Reply:
Ooh, that makes a lot of sense. I'll update the PR